newIDE/app/src/Profile/EmailVerificationPendingDialog.js

```diff
@@ -0,0 +1,90 @@
+// @flow
+import { Trans } from '@lingui/macro';
+
+import React from 'react';
+import FlatButton from '../UI/FlatButton';
+import RaisedButton from '../UI/RaisedButton';
+import Dialog from '../UI/Dialog';
+import { type AuthenticatedUser } from './AuthenticatedUserContext';
+import { Column, Line, Spacer } from '../UI/Grid';
+import CircularProgress from '@material-ui/core/CircularProgress';
+import BackgroundText from '../UI/BackgroundText';
+import VerifiedUser from '@material-ui/icons/VerifiedUser';
+import Text from '../UI/Text';
+import { useInterval } from '../Utils/UseInterval';
+
+type Props = {|
+  onClose: Function,
+  authenticatedUser: AuthenticatedUser,
+|};
+
+export default function EmailVerificationPendingDialog({
+  onClose,
+  authenticatedUser,
+}: Props) {
+  const isVerified =
+    !!authenticatedUser &&
+    !!authenticatedUser.firebaseUser &&
+    !!authenticatedUser.firebaseUser.emailVerified;
+  useInterval(
+    () => authenticatedUser.onRefreshFirebaseProfile(),
+    isVerified ? null : 3900
+  );
+
+  return (
+    <Dialog
+      actions={[
+        isVerified ? (
+          <RaisedButton
+            label={<Trans>Done!</Trans>}
+            key="close"
+            primary
+            onClick={onClose}
+          />
+        ) : (
+          <FlatButton
+            label={<Trans>Cancel and close</Trans>}
+            key="close"
+            primary={false}
+            onClick={onClose}
+          />
+        ),
+      ]}
+      title={undefined}
+      maxWidth="sm"
+      cannotBeDismissed={true}
+      open
+      noMargin
+    >
+      {!isVerified && (
```

Comment:
Uber nitpicking: use `isVerified ? (...) : (...)`? 
Might emphasize that this dialog has 2 states, no more.